src/clang.rs

```diff
@@ -287,11 +287,15 @@ impl Cursor {
 
     /// Given that this cursor points to a template specialization, get a cursor
     /// pointing to the template definition that is being specialized.
-    pub fn specialized(&self) -> Cursor {
+    pub fn specialized(&self) -> Option<Cursor> {
+        if !self.is_valid() {
+            return None;
```

Comment:
Rather than checking if `self.is_valid()`, we should make the FFI call to clang, and then check if the result of that FFI call `.is_valid()` and return `None`/`Some` based on that.

Does that make sense?